lib/SILOptimizer/Utils/Devirtualize.cpp

```diff
@@ -670,6 +670,15 @@ bool swift::canDevirtualizeClassMethod(FullApplySite AI,
       return false;
   }
 
+  // devirtualizeClassMethod below does not support this case. It currently
+  // assumes it can try_apply call the target.
+  if (!F->getLoweredFunctionType()->hasErrorResult() &&
```

Comment:
Do you need a similar check for witness methods as well? Also I think it should be relatively easy to rewrite the try_apply into an apply followed by an unconditional branch, making the error block unreachable. Can you file a bug to handle that?

Reply:
Sure

https://bugs.swift.org/browse/SR-9112

Reply:
Thanks. Can you add a test for the protocol case too?